client/src/app/pantry/pantry-products-list/pantry-products-list.component.ts

```diff
@@ -213,57 +212,19 @@ export class PantryProductsListComponent implements OnInit, OnDestroy {
   }
 
   public makeCategoryLists(): void {
-    this.bakedGoodsItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'baked goods'});
-    this.produceItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'produce'});
-    this.meatItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'meat'});
-    this.dairyItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'dairy'});
-    this.frozenItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'frozen foods'});
-    this.herbItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'herbs and spices'});
-    this.beverageItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'beverages'});
-    this.cleaningItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'cleaning products'});
-    this.paperItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'paper products'});
-    this.miscellaneousItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'miscellaneous'});
-    this.deliItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'deli'});
-    this.stapleItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'staples'});
-    this.toiletriesItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'toiletries'});
-    this.bakingSuppliesItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'baking supplies'});
-    this.petSuppliesItems = this.pantryService.filterItems(
-      this.pantryProducts, { category: 'pet supplies'});
+    categories.forEach((cat: ProductCategory) => {
+      const categoryAsField = categoryCamelCase(cat);
+      this[categoryAsField] = this.pantryService.filterItems(
```

Comment:
If `categoryAsField` is `bakedGoodsItems`, then `this[categoryAsField]` is equivalent to `this.bakedGoodsItems`.